vllm/model_executor/models/transformers/moe.py

```diff
@@ -256,7 +256,14 @@ def forward(self, *args, **kwargs):
         def _recursive_replace(module: nn.Module, prefix: str):
             for child_name, child_module in module.named_children():
                 qual_name = maybe_prefix(prefix, child_name)
-                if child_name == "experts" and isinstance(child_module, nn.ModuleList):
+                # Naive implementations will have experts as ModuleList
+                is_modulelist = isinstance(child_module, nn.ModuleList)
+                # Packed implementations will have experts as 3D tensors of shapes like:
+                # gate_up_proj = (num_experts, 2 * intermediate_size, hidden_size)
+                # down_proj = (num_experts, intermediate_size, hidden_size)
+                params = list(child_module.parameters())
+                is_3d = bool(params) and all(p.ndim == 3 for p in params)
```

Comment:
```suggestion
                is_3d = len(params) > 0 and all(p.ndim == 3 for p in params)
```

I prefer to be more explicit